var/spack/repos/builtin/packages/py-scipy/package.py

```diff
@@ -73,7 +73,7 @@ class PyScipy(PythonPackage):
     depends_on('py-numpy@1.16.5:1.22+blas+lapack', when='@1.6.2:', type=('build', 'link', 'run'))
     depends_on('py-cython@0.29.18:2', when='@1.7:', type='build')
     depends_on('py-pythran@0.9.11', when='@1.7.0:1.7.1', type=('build', 'link'))
-    depends_on('py-pythran@0.9.12:0.9', when='@1.7.2:', type=('build', 'link'))
+    depends_on('py-pythran@0.9.12:0.10.0', when='@1.7.2:', type=('build', 'link'))
```

Comment:
actualy, this isn't right either, I guess `0.9` may be your way of specifying `<0.10.0`?

Reply:
hmm, if I pin to `0.9.12` exactly that does concretize "ok" it seems. To be clear, my original error was:

```
Cannot merge constraint
py-gast@0.4.0:0.4
into
py-gast@0.5.0:0.5
```

and simply changing this pin seemed to do the trick